controllers/core/event_controller.go

```diff
@@ -0,0 +1,235 @@
+)
+
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;update;patch;list;watch
+
+type EventReconciler struct {
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+	K8sAPI k8s.K8sWrapper
+}
+
+var (
+	eventControllerTotalEventsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "total_events_reconciled_by_event_controller",
+			Help: "The number of events that were reconcile by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerSGPEventsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "security_group_pod_events_reconciled_by_event_controller",
+			Help: "The number of SGP events that were reconcile by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerSGPEventsFailureCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "failed_security_group_pod_events_reconciled_by_event_controller",
+			Help: "The number of SGP events that were reconcile by the controller but failed on label node",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerCNEventsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "custom_networking_events_reconciled_by_event_controller",
+			Help: "The number of custom networking events that were reconcile by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerCNEventsFailureCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "failed_custom_networking_events_reconciled_by_event_controller",
+			Help: "The number of custom networking events that were reconcile by the controller but failed on label node",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerEventFailureCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "failed_get_events__by_event_controller",
+			Help: "The number of failures on getting events by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventReconcileOperationLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "event_reconcile_operation_latency",
+			Help: "Event controller reconcile events latency in ms",
+		},
+		[]string{"time"},
+	)
+
+	prometheusRegistered = false
+	reconcileEvents      = "reconcile_events"
+)
+
+func PrometheusRegister() {
+	if !prometheusRegistered {
+		metrics.Registry.MustRegister(eventControllerTotalEventsCount)
+		metrics.Registry.MustRegister(eventControllerSGPEventsCount)
+		metrics.Registry.MustRegister(eventControllerEventFailureCount)
+		metrics.Registry.MustRegister(eventControllerCNEventsCount)
+		metrics.Registry.MustRegister(eventReconcileOperationLatency)
+		metrics.Registry.MustRegister(eventControllerSGPEventsFailureCount)
+		metrics.Registry.MustRegister(eventControllerCNEventsFailureCount)
+		prometheusRegistered = true
+	}
+}
+
+func (r *EventReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	start := time.Now()
+	var err error
+
+	logger := r.Log.WithValues("event", req.Name)
```

Comment:
nit: maybe we can add const for things like "event", "reconcile"..similar to this - `EventFilterKey = "reportingComponent"`